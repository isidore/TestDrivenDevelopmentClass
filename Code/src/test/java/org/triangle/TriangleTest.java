package org.triangle;

import static org.junit.Assert.assertEquals;

import java.awt.Point;

import org.approvaltests.Approvals;
import org.junit.Test;

public class TriangleTest {

	@Test
	public void testLine() {
		Line line = Line.makeLine(new Point(0, 0), new Point(0, 5));
		assertEquals(5.0, line.getLength(), 1e-4);
	}

	@Test
	public void testHorizontal() {
		Line line = Line.makeLine(new Point(0, 0), new Point(5, 0));
		assertEquals(5.0, line.getLength(), 1e-4);
	}

	@Test
	public void testDiagonal() {
		Line line = Line.makeLine(new Point(0, 0), new Point(4, 3));
		assertEquals(5.0, line.getLength(), 1e-4);
	}

	@Test
	public void testLineEndpoints() {
		Line line = Line.makeLine(new Point(2, 1), new Point(6, 5));
		Approvals.verifyAll("linepoints", line.getPoints());
	}

	@Test
	public void testTriangle() throws Exception {
		// make a new /\ with point (1,2), (5,3),(2,6)
		Triangle triangle = new Triangle(new Point(1, 2), new Point(5, 3), new Point(2, 6));
		// verify the ....
		Approvals.verifyAll("trianglePoints", triangle.getPoints());
	}

	@Test
	public void testTriangleLines() throws Exception {
		// make a new /\ with point (1,2), (5,3),(2,6)
		Triangle triangle = new Triangle(new Point(1, 2), new Point(5, 3), new Point(2, 6));
		// verify lines
		Approvals.verifyAll("triangleLines", triangle.getLines());
	}

	@Test
	public void shouldReturnLines() throws Exception {
		// make a new /\ with point (1,2), (5,3),(2,6)
		Triangle triangle = new Triangle(new Point(1, 2), new Point(5, 3), new Point(2, 6));

		// verify lines which points are (1,2)
		Approvals.verifyAll("triangleLinesByPoints", triangle.getCrossingLines(new Point(1, 2)));
	}

	@Test
	public void testTriangleState() throws Exception {
		Triangle triangle = new Triangle(new Point(1, 2), new Point(5, 3), new Point(2, 6));
		Approvals.verifyAsJson(triangle);
	}
}
