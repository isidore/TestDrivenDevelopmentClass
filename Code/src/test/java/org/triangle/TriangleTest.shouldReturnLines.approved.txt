triangleLinesByPoints[0] = line = (1.0,2.0)(5.0,3.0)
triangleLinesByPoints[1] = line = (2.0,6.0)(1.0,2.0)
